### ISSUE TITLE
LKRG build broken on Linux 5.10.220 due to find_module and module_mutex

### ISSUE THREAD

**Issue by 0strodamus** [lkrg-org/lkrg#341]
Building LKRG on Linux 5.10.220 is broken with the following errors:
```
ERROR: modpost: "find_module" [/var/lib/dkms/lkrg/0.9.8/build/lkrg.ko] undefined!
ERROR: modpost: "module_mutex" [/var/lib/dkms/lkrg/0.9.8/build/lkrg.ko] undefined!
```
This is due to commit [32edffff869a224f0c4705e4c46a0598e49cf7a6](https://github.com/gregkh/linux/commit/32edffff869a224f0c4705e4c46a0598e49cf7a6), which removes exporting find_module and module_mutex. Reverting this commit and recompiling linux-5.10.220 allowed LKRG to compile successfully.

**Comment by solardiz** [comment]
Thanks. So we need to revise the version checks introduced in 523dd2c5b07af6f6dd5116219ae6083c74d3bd9d from 5.12+ to 5.10+ or maybe 5.10.220+ - but for the latter we first need to identify the exact 5.10.y version from which this change exists, maybe it's slightly lower than 220?

Can you test this (with non-patched kernel)? -

```diff
+++ b/src/modules/kmod/p_kmod.c
@@ -38,7 +38,7 @@ int p_kmod_init(void) {
    P_SYM_INIT(modules)
    P_SYM_INIT(module_kset)
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,12,0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,10,0)
    P_SYM_INIT(module_mutex)
    P_SYM_INIT(find_module)
 #else
```

**Comment by 0strodamus** [comment]
You're welcome. I will definitely test your diff with a non-patched kernel and report back. My preference would be to wait until the next kernel release because kernel compiles aren't very fast on my system, so it may be a little while.

You can verify the 5.10.y version by looking at the [changelog](https://mirrors.edge.kernel.org/pub/linux/kernel/v5.x/ChangeLog-5.10.220) for 5.10.220 and searching for "module: unexport find_module and module_mutex". The commit is at this [location](https://github.com/gregkh/linux/commits/linux-5.10.y/?after=3a3877de44342d0a09216dfbe674a404e8f5e96f+559) about halfway down the page. Hope that helps.

Thanks for the prompt response and I'll update this issue report when I'm able to test.

**Comment by 0strodamus** [comment]
I tested your diff with a non-patched 5.10.220 kernel and it works. Thanks for the fix!

**Comment by solardiz** [comment]
Thank you @0strodamus! I've committed a revision of this change, checking for 5.10.220+.
